pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,51 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// VersionComparator enables kubernetes version retrieval.
+type VersionComparator interface {
+	GetVersion() string
+}
+
+// ReplicaComparator enables replica number retrieval.
+type ReplicaComparator interface {
```

Comment:
what kind of replicas? or are you using this for different comparisons?

Reply:
Updating naming to "Replicatable". I was using this for getting the machine replicas, whether control plane or worker node. I'll update to `MachineReplicaComparator`